ports/stm32/uart.c

```diff
@@ -497,6 +522,32 @@ bool uart_init(pyb_uart_obj_t *uart_obj,
             break;
         #endif
 
+        #if defined(MICROPY_HW_LPUART2_TX) && defined(MICROPY_HW_LPUART2_RX)
+        case PYB_LPUART_2:
+            uart_fn = AF_FN_LPUART;
+            uart_unit = 2;
+            UARTx = LPUART2;
+            #if defined(STM32G0)
+            irqn = USART2_LPUART2_IRQn;
+            #else
+            irqn = USART2_IRQn;
```

Comment:
I guess this line hasn't been tested on hardware, so best to remove it, just support the G0 for LPUART2 for now.